### ISSUE TITLE
Can't publish a new feature on next

### ISSUE THREAD

**Issue by craigsketchley** [semantic-release/semantic-release#1145]
## Current behavior

- **`master`** has a release `0.11.0`
- **`next`** branch is created with a new feature commit (`feat`).
- a `EINVALIDNEXTVERSION` error is thrown with the following reason:

```
The release `0.12.0` on branch `next` cannot be published as it is out of range.

Based on the releases published on other branches, only versions within the
range >=1.0.0 can be published from branch next.
```

## Expected behavior

- **`master`** has a release `0.11.0`
- **`next`** branch is created with a new feature commit (`feat`).
- a new release is published with version `0.12.0` from `next` branch on `next` channel.

## Environment

- **semantic-release** version: 16.0.0-beta.18
- CI environment: CircleCI & Locally
- Plugins used:
  - @semantic-release/exec: 3.4.0-beta.2
  - semantic-release-chrome: 1.1.1 (with some custom tweaks to work with release channels)
- **semantic-release** configuration:

```
{
  "plugins": [
    "@semantic-release/commit-analyzer",
    "@semantic-release/release-notes-generator",
    ["@semantic-release/npm", {
      "npmPublish": false
    }],
    ["@semantic-release/exec", {
      "prepareCmd": "./prepare.sh ${!nextRelease.channel ? 'production' : 'staging'}"
    }],
    ["semantic-release-chrome", {
      "asset": "dist.zip",
      "extensionId": {
        "default": "<extension_id1>",
        "next": "<extension_id2>"
      }
    }],
    "@semantic-release/github"
  ]
}
```

- CI logs:

```
[7:54:57 AM] [semantic-release] › ℹ  Running semantic-release version 16.0.0-beta.18
[7:54:57 AM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/npm"
[7:54:57 AM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/exec"
[7:54:57 AM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from semantic-release-chrome"
[7:54:57 AM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/github"
[7:54:57 AM] [semantic-release] › ✔  Loaded plugin "analyzeCommits" from "@semantic-release/commit-analyzer"
[7:54:57 AM] [semantic-release] › ✔  Loaded plugin "analyzeCommits" from "@semantic-release/exec"
[7:54:57 AM] [semantic-release] › ✔  Loaded plugin "verifyRelease" from "@semantic-release/exec"
[7:54:57 AM] [semantic-release] › ✔  Loaded plugin "generateNotes" from "@semantic-release/release-notes-generator"
[7:54:57 AM] [semantic-release] › ✔  Loaded plugin "generateNotes" from "@semantic-release/exec"
[7:54:57 AM] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/npm"
[7:54:57 AM] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/exec"
[7:54:57 AM] [semantic-release] › ✔  Loaded plugin "prepare" from semantic-release-chrome"
[7:54:57 AM] [semantic-release] › ✔  Loaded plugin "publish" from "@semantic-release/npm"
[7:54:57 AM] [semantic-release] › ✔  Loaded plugin "publish" from "@semantic-release/exec"
[7:54:57 AM] [semantic-release] › ✔  Loaded plugin "publish" from semantic-release-chrome"
[7:54:57 AM] [semantic-release] › ✔  Loaded plugin "publish" from "@semantic-release/github"
[7:54:57 AM] [semantic-release] › ✔  Loaded plugin "addChannel" from "@semantic-release/npm"
[7:54:57 AM] [semantic-release] › ✔  Loaded plugin "addChannel" from "@semantic-release/exec"
[7:54:57 AM] [semantic-release] › ✔  Loaded plugin "success" from "@semantic-release/exec"
[7:54:57 AM] [semantic-release] › ✔  Loaded plugin "success" from "@semantic-release/github"
[7:54:57 AM] [semantic-release] › ✔  Loaded plugin "fail" from "@semantic-release/exec"
[7:54:57 AM] [semantic-release] › ✔  Loaded plugin "fail" from "@semantic-release/github"
[7:54:59 AM] [semantic-release] › ✔  Run automated release from branch next
[7:54:59 AM] [semantic-release] › ✔  Allowed to push to the Git repository
[7:54:59 AM] [semantic-release] › ℹ  Start step "verifyConditions" of plugin "@semantic-release/npm"
[7:54:59 AM] [semantic-release] › ✔  Completed step "verifyConditions" of plugin "@semantic-release/npm"
[7:54:59 AM] [semantic-release] › ℹ  Start step "verifyConditions" of plugin "@semantic-release/exec"
[7:54:59 AM] [semantic-release] › ✔  Completed step "verifyConditions" of plugin "@semantic-release/exec"
[7:54:59 AM] [semantic-release] › ℹ  Start step "verifyConditions" of plugin semantic-release-chrome"
[7:54:59 AM] [semantic-release] › ✔  Completed step "verifyConditions" of plugin semantic-release-chrome"
[7:54:59 AM] [semantic-release] › ℹ  Start step "verifyConditions" of plugin "@semantic-release/github"
[7:54:59 AM] [semantic-release] [@semantic-release/github] › ℹ  Verify GitHub authentication
[7:55:00 AM] [semantic-release] › ✔  Completed step "verifyConditions" of plugin "@semantic-release/github"
[7:55:00 AM] [semantic-release] › ℹ  Found git tag v0.11.0 associated with version 0.11.0 on branch next
[7:55:00 AM] [semantic-release] › ℹ  Found 1 commits since last release
[7:55:00 AM] [semantic-release] › ℹ  Start step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
[7:55:00 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: feat: next dummy feature
[7:55:00 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The release type for the commit is minor
[7:55:00 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analysis of 1 commits complete: minor release
[7:55:00 AM] [semantic-release] › ✔  Completed step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
[7:55:00 AM] [semantic-release] › ℹ  Start step "analyzeCommits" of plugin "@semantic-release/exec"
[7:55:00 AM] [semantic-release] › ✔  Completed step "analyzeCommits" of plugin "@semantic-release/exec"
[7:55:00 AM] [semantic-release] › ℹ  The next release version is 0.12.0
[7:55:00 AM] [semantic-release] › ℹ  Start step "fail" of plugin "@semantic-release/exec"
[7:55:00 AM] [semantic-release] › ✔  Completed step "fail" of plugin "@semantic-release/exec"
[7:55:00 AM] [semantic-release] › ℹ  Start step "fail" of plugin "@semantic-release/github"
[7:55:03 AM] [semantic-release] [@semantic-release/github] › ℹ  Created issue #85
[7:55:03 AM] [semantic-release] › ✔  Completed step "fail" of plugin "@semantic-release/github"
[7:55:03 AM] [semantic-release] › ✖  EINVALIDNEXTVERSION The release `0.12.0` on branch `next` cannot be published as it is out of range.
Based on the releases published on other branches, only versions within the range >=1.0.0 can be published from branch next.

The following commit is responsible for the invalid release:

    * feat: next dummy feature (842ced3)

This commit should be moved to a valid branch with git merge (https://git-scm.com/docs/git-merge) or git cherry-pick (https://git-scm.com/docs/git-cherry-pick) and removed from branch next with git revert (https://git-scm.com/docs/git-revert) or git reset (https://git-scm.com/docs/git-reset).

A valid branch could be master or next.

See the workflow configuration documentation (https://github.com/semantic-release/semantic-release/blob/master/docs/usage/workflow-configuration.md) for more details.

{ SemanticReleaseError: The release `0.12.0` on branch `next` cannot be published as it is out of range.
    at module.exports (/home/circleci/repo/node_modules/semantic-release/lib/get-error.js:6:10)
    at run (/home/circleci/repo/node_modules/semantic-release/index.js:160:11)
    at process._tickCallback (internal/process/next_tick.js:68:7)
  name: 'SemanticReleaseError',
  code: 'EINVALIDNEXTVERSION',
  details:
   'Based on the releases published on other branches, only versions within the range `>=1.0.0` can be published from branch `next`.\n\nThe following commit is responsible for the invalid release:\n- feat: next dummy feature (842ced3)\n\nThis commit should be moved to a valid branch with [git merge](https://git-scm.com/docs/git-merge) or [git cherry-pick](https://git-scm.com/docs/git-cherry-pick) and removed from branch `next` with [git revert](https://git-scm.com/docs/git-revert) or [git reset](https://git-scm.com/docs/git-reset).\n\nA valid branch could be `master` or `next`.\n\nSee the [workflow configuration documentation](https://github.com/semantic-release/semantic-release/blob/master/docs/usage/workflow-configuration.md) for more details.',
  semanticRelease: true }npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! test-project@0.0.0 release: `semantic-release`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the test-project@0.0.0 release script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/circleci/.npm/_logs/2019-04-10T07_55_03_367Z-debug.log
Exited with code 1
```

**Comment by craigsketchley** [comment]
Apologies if I've missed something but I'm trying to set up two release branches; one for production off `master` and one for staging/internal testing off `next`. I'd need to include both features (minor) and breaking changes (major) in the `next` branch releases - and when we're happy - merge them into `master`. Any help would be greatly appreciated.

**Comment by craigsketchley** [comment]
I did some more digging and it looks like if I create a third branch (e.g. `next-major`), then I can have minor changes in `next` - but not if I only have two branches. You're not allowing minor changes on other branches if there are only two (or less) matching branches in git (the line is [here](https://github.com/semantic-release/semantic-release/blob/d10268a8fa6a76967ea07ea2473dee32fd953cfb/lib/branches/normalize.js#L64) in `lib/branches/normalize.js`).

I guess I'm after some way to specify the level of changes (major and/or minor) available for a secondary branch. Does something like that align with your plans? Something I can contribute?

**Comment by travi** [comment]
i'm not familiar enough with that part of the implementation to say for sure, but i would suspect that it is likely related to the fact that you are using version numbers that are lower than `1.0.0`. that is not officially supported by semantic-release: https://semantic-release.gitbook.io/semantic-release/support/faq#can-i-set-the-initial-release-version-of-my-package-to-0-0-1

**Comment by craigsketchley** [comment]
Thanks @travi. I thought that too but it didn’t change the behaviour when I had master on a >1.0.0 version.

**Comment by craigsketchley** [comment]
I guess the problem for me is in the `release()` function in `lib/branches/normalize.js`. I'll mock out what my current and expected behaviour of that function is...

## Input

Given the following input to the `release()` function in `lib/branches/normalize.js`:

```
{
  release: [
    {
      name: 'master',
      tags: [
        {
          version: '1.0.0',
          channel: undefined,
          gitTag: 'v1.0.0',
          gitHead: 'SOME_GITHASH'
        }
      ]
    },
    {
      name: 'next',
      tags: [
        {
          version: '1.0.0',
          channel: undefined,
          gitTag: 'v1.0.0',
          gitHead: 'SOME_GITHASH'
        }
      ]
    }
  ]
}
```

## Current Output

The current output gives:

```
[
  {
    name: 'master',
    channel: undefined,
    tags: [
      {
        version: '1.0.0',
        channel: undefined,
        gitTag: 'v1.0.0',
        gitHead: 'SOME_GITHASH'
      }
    ],
    type: 'release',
    range: '>=1.0.0 <2.0.0',
    accept: ['patch', 'minor']
  },
  {
    name: 'next',
    channel: 'next',
    tags: [
      {
        version: '1.0.0',
        channel: undefined,
        gitTag: 'v1.0.0',
        gitHead: 'SOME_GITHASH'
      }
    ],
    type: 'release',
    range: '>=2.0.0',
    accept: ['major']
  }
]
```

## Expected Output

My expected output:

```
[
  {
    name: 'master',
    channel: undefined,
    tags: [
      {
        version: '1.0.0',
        channel: undefined,
        gitTag: 'v1.0.0',
        gitHead: 'SOME_GITHASH'
      }
    ],
    type: 'release',
    range: '>=1.0.0 <1.1.0',
    accept: ['patch']
  },
  {
    name: 'next',
    channel: 'next',
    tags: [
      {
        version: '1.0.0',
        channel: undefined,
        gitTag: 'v1.0.0',
        gitHead: 'SOME_GITHASH'
      }
    ],
    type: 'release',
    range: '>=1.1.0',
    accept: ['minor', 'major']
  }
]
```

**Notice the change in `range` and `accept` properties.**

Either this should happen by default or we should at least have the option to enable this behaviour in the config (quite what that looks like - I'm not sure).

**Comment by julrich** [comment]
@craigsketchley Did you find a workaround for this? Currently running into the same thing (I think, at least), on `16.0.0-beta.24`.

Fresh 1.0.0 releases on `master`, `next` and `next-major`. The following / first feature-Release on `next` is blocked with:
```
Based on the releases published on other branches, only versions within the range `>=2.0.0` can be published from branch `next`.\n'
```

**Comment by craigsketchley** [comment]
No, sorry @julrich. I actually stopped using semantic-release. I figured I'd revisit when v16 was officially released. I think I was having trouble because I was trying to use SR for app versioning/deployments, and not package releases.

**Comment by ellisio** [comment]
I am also running into this issue on our build trying to publish `v2.1.0` on `next`. Here is the current `git log` output:

```
* af54c28 (origin/next, next) feat(textfield): add the TextField component (#100)
* 6f9a25c (tag: v2.0.0@next) feat: improve extensability of components
| * d537485 (HEAD -> master, origin/master, origin/HEAD) build(deps-dev): bump lint-staged from 9.2.5 to 9.3.0 (#99)
|/  
* c0a59a1 build(deps-dev): bump @storybook/addon-actions from 5.2.0 to 5.2.1 (#97)
```

The current tag on `master` is `v1.3.0` which is back several commits on `master`. However, I would assume this should have no impact on a feature being released on `next`.

This is the CI output:

```
[10:30:38 PM] [semantic-release] › ✖  EINVALIDNEXTVERSION The release `2.1.0` on branch `next` cannot be published as it is out of range.
Based on the releases published on other branches, only versions within the range >=3.0.0 can be published from branch next.

The following commit is responsible for the invalid release:

    * feat(textfield): add the TextField component (#100) (af54c28)

This commit should be moved to a valid branch with git merge (https://git-scm.com/docs/git-merge) or git cherry-pick (https://git-scm.com/docs/git-cherry-pick) and removed from branch next with git revert (https://git-scm.com/docs/git-revert) or git reset (https://git-scm.com/docs/git-reset).

A valid branch could be master or next.

See the workflow configuration documentation (https://github.com/semantic-release/semantic-release/blob/master/docs/usage/workflow-configuration.md) for more details.

SemanticReleaseError: The release `2.1.0` on branch `next` cannot be published as it is out of range.
```

**Comment by julrich** [comment]
Btw., it seems to me even the error message is approving of the "violating" commit being put on `master` or `next`:
```
A valid branch could be master or next.
``` 

Will probably try to debug this!

Packages used by us:
```
"semantic-release": "16.0.0-beta.24",
"@semantic-release/changelog": "^3.0.4",
"@semantic-release/git": "7.1.0-beta.3",
```

**Comment by julrich** [comment]
Okay, so I've fixed our current problem. The restriction seems to be in effect if you're (like us) trying to use `next` and `next-major` branches, with the `prerelease' flag set for `next-major`. This flag results in `next-major` not counting as a `release`-branch anymore, which in turn results in the restrictions for the `next` + `master` combination going into effect.

We just removed the `prerelease`-flag, as it was just a test anyways. Maybe this should be mentioned in the docs somewhere? Not sure about where, though.

**Comment by pvdlg** [comment]
Sorry for the late answer. Please see https://github.com/semantic-release/semantic-release/blob/beta/docs/usage/workflow-configuration.md#pushing-to-a-release-branch

From the description above it seems to work as expected. With a 2 release branch configuration, `master` is expected to receive `patch` and `minor` releases while `next` is expected to receive the next `major`.
In other works if the last release on `master` was `1.0.0` and `next` is on par with `master`:
- `master` should receive `fix` and `feat` commits but no `BREAKING CHANGE`
- `next` should receive at least one `BREAKING CHANGE` commit then can receive any other commit

**Comment by pvdlg** [comment]
I'm closing as it seems to work as expected. If it doesn't please reopen.

**Comment by ellisio** [comment]
@pvdlg Then is this documentation "recipe" wrong?

https://github.com/semantic-release/semantic-release/blob/beta/docs/recipes/distribution-channels.md#publishing-on-distribution-channels

We are using the default configuration, so I would presume having `>= 1.0.0 < 2.0.0` on `master` would work. Then `next` being `>= 2.0.0` with `feat`, `fix` being pushable to release `minor` and `fix` releases on `next` until `next` is ready to be merged to `master`.

**Comment by pvdlg** [comment]
Where is it incorrect?

**Comment by ellisio** [comment]
@pvdlg The language in that recipe makes it sound like the following workflow would work; which it is not, and the error I think OP and myself are running into.

1. Create a release on `master`, v1.0.0.
2. Create a branch off `master`, with `next`, and commit a `BREAKING CHANGE`; which results in `v2.0.0`.
3. Make a fix to `master`, resulting in `v1.0.1` on `master`.
4. Make a feature on `next`, resulting in `v2.1.0` on `next`.
5. Merge the fix from `master` into `next` as it's unrelated to the development on `next`, but is a security fix. This should result in `v2.1.1` on `next`.
6. Merge `next` into `master`, causing `v2.1.1` to get promoted onto the `@latest` NPM channel.

Under this workflow, it breaks at step 4 with the errors described above. The recipe makes it sound like this should be possible.

Is `next` simply for `v2.0.x` in this pattern? Are we supposed to be merging `v2.0.x` into `master` (`@latest` distro) before working on another feature in the `v2.x.x` release? If this is the case, I think the above recipe could use a documentation block explaining that behavior when using only `master` and `next`. Could prevent maintainers from doing the wrong thing, and perhaps link them to using `master`, `next`, `next-major` patterns instead.

**Comment by ellisio** [comment]
@pvdlg I think this is a CI bug. I just tested with a stub repository to validate what I'm doing and ran into a weird behavior.

Locally on my `next` branch, I commit a `fix: something` commit to bump to `v2.0.1`. This results in the following history:

```
* 3a18810 - (HEAD -> next, origin/next) fix: test a fix (4 minutes ago) <Andrew Ellis>
* 78e4e2e - (tag: v2.0.0@next) feat: doing something (5 minutes ago) <Andrew Ellis>
* 9ed2672 - (tag: v1.0.0, origin/master, master) chore: fix plugins config (15 minutes ago) <Andrew Ellis>
* e382f07 - feat: initial commit :tada: (17 minutes ago) <Andrew Ellis>
```

When I run `semantic-release` locally to do a dry run I get the following output:

```
$ ✔ npm run semantic-release

> test@0.0.0-development semantic-release /Users/andrew/Developer/test
> semantic-release

[12:04:13 PM] [semantic-release] › ℹ  Running semantic-release version 16.0.0-beta.24
[12:04:13 PM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/github"
[12:04:13 PM] [semantic-release] › ✔  Loaded plugin "analyzeCommits" from "@semantic-release/commit-analyzer"
[12:04:13 PM] [semantic-release] › ✔  Loaded plugin "generateNotes" from "@semantic-release/release-notes-generator"
[12:04:13 PM] [semantic-release] › ✔  Loaded plugin "publish" from "@semantic-release/github"
[12:04:13 PM] [semantic-release] › ✔  Loaded plugin "addChannel" from "@semantic-release/github"
[12:04:13 PM] [semantic-release] › ✔  Loaded plugin "success" from "@semantic-release/github"
[12:04:13 PM] [semantic-release] › ✔  Loaded plugin "fail" from "@semantic-release/github"
[12:04:13 PM] [semantic-release] › ⚠  This run was not triggered in a known CI environment, running in dry-run mode.
[12:04:15 PM] [semantic-release] › ⚠  Run automated release from branch next in dry-run mode
[12:04:15 PM] [semantic-release] › ✔  Allowed to push to the Git repository
[12:04:15 PM] [semantic-release] › ℹ  Start step "verifyConditions" of plugin "@semantic-release/github"
[12:04:15 PM] [semantic-release] [@semantic-release/github] › ℹ  Verify GitHub authentication
[12:04:16 PM] [semantic-release] › ✔  Completed step "verifyConditions" of plugin "@semantic-release/github"
[12:04:16 PM] [semantic-release] › ℹ  Found git tag v2.0.0@next associated with version 2.0.0 on branch next
[12:04:16 PM] [semantic-release] › ℹ  Found 1 commits since last release
[12:04:16 PM] [semantic-release] › ℹ  Start step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
[12:04:16 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: fix: test a fix
[12:04:16 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The release type for the commit is patch
[12:04:16 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analysis of 1 commits complete: patch release
[12:04:16 PM] [semantic-release] › ✔  Completed step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
[12:04:16 PM] [semantic-release] › ℹ  The next release version is 2.0.1
[12:04:16 PM] [semantic-release] › ℹ  Start step "generateNotes" of plugin "@semantic-release/release-notes-generator"
[12:04:16 PM] [semantic-release] › ✔  Completed step "generateNotes" of plugin "@semantic-release/release-notes-generator"
[12:04:16 PM] [semantic-release] › ⚠  Skip v2.0.1@next tag creation in dry-run mode
[12:04:16 PM] [semantic-release] › ⚠  Skip step "publish" of plugin "@semantic-release/github" in dry-run mode
[12:04:16 PM] [semantic-release] › ⚠  Skip step "success" of plugin "@semantic-release/github" in dry-run mode
[12:04:16 PM] [semantic-release] › ✔  Published release 2.0.1
[12:04:16 PM] [semantic-release] › ℹ  Release note for version 2.0.1:
## 2.0.1 (https://github.com/salesmessage/test/compare/v2.0.0@next...v2.0.1@next) (2019-09-27)

### Bug Fixes

    * test a fix (3a18810 (https://github.com/salesmessage/test/commit/3a18810))
```

Sweet, should work when pushed to CI, right!? Nope, not so fast. This is the output on CircleCI:

```
[6:04:58 PM] [semantic-release] › ℹ  Running semantic-release version 16.0.0-beta.24
[6:04:58 PM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/github"
[6:04:58 PM] [semantic-release] › ✔  Loaded plugin "analyzeCommits" from "@semantic-release/commit-analyzer"
[6:04:58 PM] [semantic-release] › ✔  Loaded plugin "generateNotes" from "@semantic-release/release-notes-generator"
[6:04:58 PM] [semantic-release] › ✔  Loaded plugin "publish" from "@semantic-release/github"
[6:04:58 PM] [semantic-release] › ✔  Loaded plugin "addChannel" from "@semantic-release/github"
[6:04:58 PM] [semantic-release] › ✔  Loaded plugin "success" from "@semantic-release/github"
[6:04:58 PM] [semantic-release] › ✔  Loaded plugin "fail" from "@semantic-release/github"
[6:04:58 PM] [semantic-release] › ✔  Run automated release from branch next
[6:04:59 PM] [semantic-release] › ✔  Allowed to push to the Git repository
[6:04:59 PM] [semantic-release] › ℹ  Start step "verifyConditions" of plugin "@semantic-release/github"
[6:04:59 PM] [semantic-release] [@semantic-release/github] › ℹ  Verify GitHub authentication
[6:04:59 PM] [semantic-release] › ✔  Completed step "verifyConditions" of plugin "@semantic-release/github"
[6:04:59 PM] [semantic-release] › ℹ  Found git tag v2.0.0@next associated with version 2.0.0 on branch next
[6:04:59 PM] [semantic-release] › ℹ  Found 1 commits since last release
[6:04:59 PM] [semantic-release] › ℹ  Start step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
[6:04:59 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: fix: test a fix
[6:04:59 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The release type for the commit is patch
[6:04:59 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analysis of 1 commits complete: patch release
[6:04:59 PM] [semantic-release] › ✔  Completed step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
[6:04:59 PM] [semantic-release] › ℹ  The next release version is 2.0.1
[6:04:59 PM] [semantic-release] › ℹ  Start step "fail" of plugin "@semantic-release/github"
[6:05:02 PM] [semantic-release] [@semantic-release/github] › ℹ  Created issue #1: https://github.com/salesmessage/test/issues/1.
[6:05:02 PM] [semantic-release] › ✔  Completed step "fail" of plugin "@semantic-release/github"
[6:05:02 PM] [semantic-release] › ✖  EINVALIDNEXTVERSION The release `2.0.1` on branch `next` cannot be published as it is out of range.
Based on the releases published on other branches, only versions within the range >=3.0.0 can be published from branch next.

The following commit is responsible for the invalid release:

    * fix: test a fix (3a18810)

This commit should be moved to a valid branch with git merge (https://git-scm.com/docs/git-merge) or git cherry-pick (https://git-scm.com/docs/git-cherry-pick) and removed from branch next with git revert (https://git-scm.com/docs/git-revert) or git reset (https://git-scm.com/docs/git-reset).

A valid branch could be master or next.

See the workflow configuration documentation (https://github.com/semantic-release/semantic-release/blob/master/docs/usage/workflow-configuration.md) for more details.

SemanticReleaseError: The release `2.0.1` on branch `next` cannot be published as it is out of range.
```

Modifying my CI to give me the Git history on CI outputs the following:

```
* 52c95b7 - (HEAD -> next, origin/next, master) fix: test a fix (38 seconds ago) <Andrew Ellis>
* 78e4e2e - (tag: v2.0.0@next) feat: doing something (9 minutes ago) <Andrew Ellis>
* 9ed2672 - (tag: v1.0.0, origin/master, origin/HEAD) chore: fix plugins config (19 minutes ago) <Andrew Ellis>
* e382f07 - feat: initial commit :tada: (21 minutes ago) <Andrew Ellis>
```

So, somewhere in CircleCI's `checkout` helper it's checking out `master` with the same commit as `next`. Which, tells me that indeed this _is not_ broken on `semantic-release`'s side. But it's a side effect of CircleCI.

To fix this, we need to reset `master` in CI to `origin/master`. Is this something that should be done in `semantic-release` when running on CircleCI?
